packages/i18n/src/create-i18n.js

```diff
@@ -22,19 +22,69 @@ const DEFAULT_LOCALE_DATA = {
 	},
 };
 
+/* eslint-disable jsdoc/valid-types */
+/**
+ * @typedef {(data?: LocaleData, domain?: string)=>void} SetLocaleData
```

Comment:
Just a formatting nit:

```suggestion
 * @typedef {(data?: LocaleData, domain?: string) => void} SetLocaleData
```

Reply:
This is another downside of using JavaScript + JSDoc + TypeScript tooling instead of TypeScript the language. Autoformatting tools don't operate on JSDoc comments, but they would on type annotations in the program itself!